### PR TITLE
Implements parallel attention/MLP

### PR DESCRIPTION
**Pull request by rodrigonogueira4**
This is the same used in GPT-J, NeoX and PALM. 
According to PALM:
"The parallel formulation results in roughly 15% faster training speed at large scales, since the MLP and Attention input matrix multiplications can be fused. Ablation experiments showed a small quality degradation at 8B scale but no quality degradation at 62B scale, so we extrapolated that the effect of parallel layers should be quality neutral at the 540B scale."

@hugoabonizio, @piresramon and @ZanezZephyrs, could you please double check this is correct? I used these implementations as reference:

GPT-J: https://github.com/kingoflolz/mesh-transformer-jax/blob/9d15698694a2a54ff0a32ec5ae4ec05e7a388bee/mesh_transformer/layers.py#L313
HF: https://github.com/huggingface/transformers/blob/94b3f544a1f5e04b78d87a2ae32a7ac252e22e31/src/transformers/models/gptj/modeling_gptj.py#L322 

Also adds instructions to use a smaller decoder-only model for testing.

### CODE REVIEW

**Review by piresramon — APPROVED** [review]

**Review by ZanezZephyrs — APPROVED** [review]
Eu rodei esse codigo na tpu v2-512, rodou certinho e trouxe um bom ganho em performance (125k tokens/s -> 139k tokens/s).

A implementação me parece certinha também.

File: t5x/examples/decoder_only/network.py [hunk 1]
```diff
@@ -95,26 +95,25 @@ def __call__(self,
...
+    mlp_output = layers.MlpBlock(
         intermediate_dim=cfg.mlp_dim,
         activations=cfg.mlp_activations,
         intermediate_dropout_rate=cfg.dropout_rate,
         dtype=cfg.dtype,
         name='mlp',
-    )(y, deterministic=deterministic)
-    y = nn.Dropout(
+    )(x, deterministic=deterministic)
+    mlp_output = nn.Dropout(
```
Comment (hugoabonizio) [inline t5x/examples/decoder_only/network.py:112]: It looks like the original mesh-transformer-jax implementation don't have this dropout while HF implements it inside [GPTJMLP](https://github.com/huggingface/transformers/blob/94b3f544a1f5e04b78d87a2ae32a7ac252e22e31/src/transformers/models/gptj/modeling_gptj.py#L272). Maybe I'm missing something?

File: README_TPU_VM.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ gcloud alpha compute tpus tpu-vm ssh $TPU_NAME --zone=$ZONE
 ```bash
 pip install "jax[tpu]>=0.2.16" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
 
-git clone --branch=main https://github.com/google-research/t5x
+git clone --branch=rodrigo-patch-5 https://github.com/rodrigonogueira4/t5x
```
Comment (hugoabonizio) [inline README_TPU_VM.md:30]: I think this will break after merging this PR

File: t5x/examples/decoder_only/network.py (hunk 1 above)
Comment (rodrigonogueira4) [reply t5x/examples/decoder_only/network.py:112]: Since dropout is disabled during pretraining, it won't hurt if we leave this code here? I mean, maybe someone will want to finetune our model in the future using dropout

File: README_TPU_VM.md (hunk 2 above)
Comment (rodrigonogueira4) [reply README_TPU_VM.md:30]: Thanks, my mistake

Comment (rodrigonogueira4) [reply README_TPU_VM.md:30]: Fixed

File: t5x/examples/decoder_only/network.py (hunk 1 above)
Comment (hugoabonizio) [reply t5x/examples/decoder_only/network.py:112]: Good catch! I'd forgotten about that!

**Review by hugoabonizio — APPROVED** [review]
